ProjectGroupA/src/main/java/com/fundation/search/controller/Validator.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * @(#)AssetFile.java Copyright (c) 2018 Jalasoft.
+ *  2643 Av Melchor Perez de Olguin, Colquiri Sud, Cochabamba, Bolivia.
+ *  All rights reserved.
+ *  <p>
+ *  This software is the confidential and proprietary information of
+ *  Jalasoft, ("Confidential Information").  You shall not
+ *  disclose such Confidential Information and shall use it only in
+ *  accordance with the terms of the license agreement you entered into
+ *  with Jalasoft.
+ *******************************************************************************/
+package com.fundation.search.controller;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+/**
+ * This class is in charge to make validations to the values for path, file name and extension.
+ *
+ * @author Jacqueline Rosales
+ * @version 1.0.
+ */
+public class Validator {
+	private Map<Integer, String> extMap;
+	/**
+     *  method in charge to send the file object to be validated.*/
+	public void validate(SearcherFile file) throws CustomSearchException {
+		validatePath(file);
+		validateFileName(file);
+		//validateExtension(file);
+	} 
+	/**
+     *  method in charge to validate the path value inserted.*/
+	private void validatePath(SearcherFile file) throws CustomSearchException {
+		File dir = new File(file.path);
+		if (!dir.exists()) { 
+			throw new CustomSearchException("The path inserted does not exist.");
+			}
+			else if (!dir.isDirectory()){
+				throw new CustomSearchException("The path inserted is not valid directory.");	
+			}
+	}
+	/**
+     *  method in charge to validate the file name value inserted.*/
+	private void validateFileName(SearcherFile file) throws CustomSearchException {
+		if (file.fileName.length() >50) {
+			throw new CustomSearchException("Your file name inserted exceeds the limit of letters allowed 50");
+			}
+			else if (checkSymbols(file.fileName)) {
+				throw new CustomSearchException("Your file name can't contain any of following characters: \\/:*?\"<>");
+			}
+	}
+	/**
+     *  method in charge to validate the extension value inserted.*/
+	private void validateExtension(SearcherFile file) throws CustomSearchException {
+		extensionsAllowed();
+		if (!extMap.containsValue(file.ext))
+			throw new CustomSearchException("The extension inserted does not exist.");
+	}
+	
+	private void extensionsAllowed() {
+		Map<Integer, String> extMap = new HashMap<Integer, String>();
```

Comment:
As reviewed on Monday, the validation should consider:
- a file might not have an extension.
- the extension might have more then 3 characters, for example "Manifest"
- we would need to validate special characters (the ones supported by Windows)